src/cachesim/cachegraphic.cpp

```diff
@@ -98,27 +98,47 @@ void CacheGraphic::updateLineReplFields(unsigned lineIdx) {
     return;
   }
 
-  if (m_cacheTextItems.at(0).at(0).lru == nullptr) {
+  if (m_cacheTextItems.at(0).at(0).lru == nullptr && m_cache.getReplacementPolicy() == ReplPolicy::LRU) {
     // The current cache configuration does not have any replacement field
     return;
   }
-
-  for (const auto &way : m_cacheTextItems[lineIdx]) {
-    // If LRU was just initialized, the actual (software) LRU value may be very
-    // large. Mask to the number of actual LRU bits.
-    unsigned lruVal = cacheLine->at(way.first).lru;
-    lruVal &= vsrtl::generateBitmask(m_cache.getWaysBits());
-    const QString lruText = QString::number(lruVal);
-    way.second.lru->setText(lruText);
-
-    // LRU text might have changed; update LRU field position to center in
-    // column
-    const qreal y = lineIdx * m_lineHeight + way.first * m_setHeight;
-    const qreal x =
-        m_widthBeforeLRU + m_lruWidth / 2 - m_fm.horizontalAdvance(lruText) / 2;
-
-    way.second.lru->setPos(x, y);
+  if (m_cacheTextItems.at(0).at(0).fifo == nullptr && m_cache.getReplacementPolicy() == ReplPolicy::FIFO){
+    return;
+  }
+  if(m_cache.getReplacementPolicy() == ReplPolicy::LRU){
+    for (const auto &way : m_cacheTextItems[lineIdx]) {
+      // If LRU was just initialized, the actual (software) LRU value may be very
+      // large. Mask to the number of actual LRU bits.
+      unsigned lruVal = cacheLine->at(way.first).lru;
+      lruVal &= vsrtl::generateBitmask(m_cache.getWaysBits());
+      const QString lruText = QString::number(lruVal);
+      way.second.lru->setText(lruText);
+
+      // LRU text might have changed; update LRU field position to center in
+      // column
+      const qreal y = lineIdx * m_lineHeight + way.first * m_setHeight;
+      const qreal x =
+          m_widthBeforeLRU + m_lruWidth / 2 - m_fm.horizontalAdvance(lruText) / 2;
+      way.second.lru->setPos(x, y);
+    }
+  }
+  if(m_cache.getReplacementPolicy() == ReplPolicy::FIFO){
+    for (const auto &way : m_cacheTextItems[lineIdx]) {
+      // If LRU was just initialized, the actual (software) LRU value may be very
+      // large. Mask to the number of actual LRU bits.
+      int fifoVal = cacheLine->at(way.first).fifo;
+      const QString fifoText = QString::number(fifoVal);
+      way.second.fifo->setText(fifoText);
+
+      // LRU text might have changed; update LRU field position to center in
+      // column
+      const qreal y = lineIdx * m_lineHeight + way.first * m_setHeight;
+      const qreal x =
+          m_widthBeforefifo + m_fifoWidth / 2 - m_fm.horizontalAdvance(fifoText) / 2;
+      way.second.fifo->setPos(x, y);
+    }
```

Comment:
LRU and FIFO code here is now essentially identical - please think about how to reduce code duplication here.